core/src/main/java/brooklyn/entity/group/DynamicClusterImpl.java

```diff
@@ -588,16 +588,22 @@ protected Entity replaceMember(Entity member, Location memberLoc, Map<?, ?> extr
 
         // choose locations to be deployed to
         List<Location> chosenLocations;
-        if (isAvailabilityZoneEnabled()) {
-            List<Location> subLocations = getNonFailedSubLocations();
-            Multimap<Location, Entity> membersByLocation = getMembersByLocation();
-            chosenLocations = getZonePlacementStrategy().locationsForAdditions(membersByLocation, subLocations, delta);
-            if (chosenLocations.size() != delta) {
-                throw new IllegalStateException("Node placement strategy chose " + Iterables.size(chosenLocations)
-                        + ", when expected delta " + delta + " in " + this);
+        chosenLocations = getMemberSpec() == null ? null : getMemberSpec().getLocations();
```

Comment:
We need to think about this more. Feels wrong to disable the availability zone behaviour entirely.

Similarly would prefer us not to merge with `FIXME: Tidy this up!` in the code. That at least should be fleshed out to a TODO that describes how it could be improved. Or ideally improve the code.